transport/src/main/java/io/netty/channel/DefaultChannelHandlerContext.java

```diff
@@ -817,8 +803,10 @@ public ChannelFuture writeAndFlush(Object msg) {
         return writeAndFlush(msg, newPromise());
     }
 
-    private static void notifyOutboundHandlerException(Throwable cause, ChannelPromise promise) {
-        PromiseNotificationUtil.tryFailure(promise, cause, logger);
+    private void handleOutboundHandlerException(Throwable cause) {
+        logger.warn("{} threw an exception while handling an outbound event." +
+                " This is most likely a bug, closing the channel.", handler(), cause);
+        close();
```

Comment:
Can this trigger an infinite loop?  (Github hides most of the file from me)

Reply:
nope as close() would just forward to the next handler

Reply:
Followup: why does the close start from the handler rather than from the end of the pipeline?

Reply:
I thought it makes more sense to start from the handler but maybe you are right and we should just start from the end of the pipeline. In this case we would need handle exceptions during close() special tho. That said I think starting from the end of the pipeline might be better as this would ensure we at least have the close propagate through all handlers. Let me do this

Reply:
Done in 730c9b9